nwb_conversion_tools/json_schema_utils.py

```diff
@@ -53,33 +54,45 @@ def get_schema_from_method_signature(class_method, exclude=None):
 
     """
     if exclude is None:
-        exclude = []
+        exclude = ['self']
+    else:
+        exclude = exclude + ['self']
     input_schema = get_base_schema()
-    annotation_json_type_map = dict(
-        bool="boolean",
-        str="string",
-        int="number",
-        float="number",
-        dict="object",
-        list="array"
-    )
-
-    for param in inspect.signature(class_method).parameters.values():
-        if param.name not in exclude + ['self']:
+    annotation_json_type_map = {
+        bool: "boolean",
+        str: "string",
+        int: "number",
+        float: "number",
+        dict: "object",
+        list: "array"
+    }
+
+    for param_name, param in inspect.signature(class_method).parameters.items():
+        if param_name not in exclude:
             if param.annotation:
-                param_type = annotation_json_type_map[str(param.annotation).split("'")[1]]
+                valid_args = [x in annotation_json_type_map for x in param.annotation.__args__]
+                if any(valid_args):
+                    param_type = [annotation_json_type_map[x] for x in np.array(param.annotation.__args__)[valid_args]]
+                else:
+                    raise ValueError("There must be only one valid annotation type that maps to json! "
+                                     f"{param.annotation.__args__} found.")
+
+                if len(set(param_type)) > 1:
+                    raise ValueError("Conflicting json parameter types were detected from the annotation! "
+                                     f"{param.annotation.__args__} found.")
```

Comment:
It's really easy at this point to test if multiple unique (i.e., conflicting) `json` output types were detected, which would be a problem.